### ISSUE TITLE
training faster-rcnn using resnet but the test detection seems not right

### ISSUE THREAD

**Issue by Trangle** [apache/incubator-mxnet#3852]
When I  tried to apply resnet(50, 101) to faster-rcnn，it works during the training process, but seemed to fail in the testing process.

the symbol was like this:

**symbol_resnet_full.py**
[symbol_resnet_full.txt](https://github.com/dmlc/mxnet/files/594099/symbol_resnet_full.txt)

**symbol_resnet.py**
[symbol_resnet.txt](https://github.com/dmlc/mxnet/files/594100/symbol_resnet.txt)

and I trained it using e2e, changed code `sym = get_resnet(num_classes=num_classes)` in line 35.

the accuracy during training process at epoch 8 of rcnn was about 94%，and rpn about 98%，but the test result was really low with mean ap 0.22，even at the training set. I used resnet-50 pretrained model downloaded from dmlc as conv initial.

**Comment by sxjscience** [comment]
@precedenceguo @tornadomeet

**Comment by precedenceguo** [comment]
I would refer you to https://github.com/tornadomeet/mx-rcnn/blob/master/rcnn/resnet.py for a good example to apply resnet to faster rcnn.
The reason why few attempts were made to resnet is that we need to run coco to verify the effectiveness, which needs too much time......

**Comment by Trangle** [comment]
@precedenceguo， I had tried it before，but the same problem occured.

**Comment by KeyKy** [comment]
@Trangle do you solve the problem? I also going to try resnet-50 with mxnet rcnn.

**Comment by Trangle** [comment]
@KeyKy Not yet, but problem was located to bn in networks!

**Comment by KeyKy** [comment]
@Trangle I found your bn1 in get_resnet_rcnn_test:
bn1 = mx.sym.BatchNorm(data=last_conv, fix_gamma=False, eps=2e-5, momentum=0.9, use_global_stats=False, name='bn1')
I think in test phase the use_global_stats should be set to True.

**Comment by Trangle** [comment]
@KeyKy debug the aux output, the problem is caused by the small batchsize limited by this frame. could you experiment it and paste your solution? I cannot solve it yet!

**Comment by KeyKy** [comment]
@Trangle I haven't got enough gpus resource right now. As soon as the new gpus servers settled down, i will try it. By the way, are you Chinese, we can use alternative communication tools such as QQ.

**Comment by Trangle** [comment]
@KeyKy you may add my QQ: 9185081

**Comment by abner2015** [comment]
do you solve the problem?
After trained  resnet, running demo.py tested the image that it did not work.
however , i had changed demo.py :
   (1) from rcnn.symbol import get_resnet_test
   (2)symbol = get_resnet_test(num_classes=config.NUM_CLASSES, num_anchors=config.NUM_ANCHORS)

**Comment by szha** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to ping me to reopen if this is still an active issue. Thanks!

**Comment by 315386775** [comment]
do you solve the problem ?  i also encounter it.

**Comment by 315386775** [comment]
# Another test
i use the pretrained model in the readme as following:
Faster R-CNN end-to-end | ResNet-101 | VOC07+12 | VOC07test | 76.4 | 79.35 | Dropbox
Use the rcnn/symbol/symbol_resnet.py.

# Commond:
python demo.py --prefix model/resnet-101 --epoch 00 --vis

# Results
I got nothing. 
INFO:root:---class---
INFO:root:[[x1, x2, y1, y2, confidence]]
